### ISSUE TITLE
Rule array-items does not support prefixItems keyword

### ISSUE THREAD

**Issue by yethee** [stoplightio/spectral#2646]
**Describe the bug**
OpenAPI 3.1 inherits [data types from JSON Schema 2020-12](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#data-types), where we can use [`prefixItems`](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-10.3.1.1) keyword to define tuple value, instead of `items` keyword.
After changes from #2638 we got an error, when schema contains `prefixItems`:

```
Schemas with "type: array", require a sibling "items" field
```

**To Reproduce**

`test.yaml`:
```yaml
---
openapi: "3.1.0"
components:
  schemas:
    IssueWithPrefixItems:
      type: array
      prefixItems:
        - type: string
        - type: integer
      minItems: 2
      maxItems: 2
      additionalItems: false
```
Run validation:
```
npx @stoplight/spectral-cli lint test.yaml

5:26    error  array-items            Schemas with "type: array", require a sibling "items" field  components.schemas.IssueWithPrefixItems
```

**Expected behavior**
Field `items` should not be required when `prefixItems` field is defined.

**Environment (remove any that are not applicable):**
```
npm ls @stoplight/spectral-rulesets

`-- @stoplight/spectral-cli@6.11.1
  +-- @stoplight/spectral-ruleset-bundler@1.5.2
  | `-- @stoplight/spectral-rulesets@1.19.1 deduped
  `-- @stoplight/spectral-rulesets@1.19.1
```

**Additional context**
This issue related to #2638.

**Comment by ponelat** [comment]
@mnaumanali94 this sounds like  a bug. Where OAS 3.1, supports `prefixItems`, `items`, `unevaluatedItems` or `contains`  (more?).

**Comment by acdha** [comment]
I just encountered this on a FastAPI project, which generates OpenAPI 3.1 schemas by default and uses `prefixItems` for any value defined as a tuple. That means that the kind of code a Python developer would normally write like this:

```python
SortItem = tuple[str, SortDirection]
…
    sort: list[SortItem] | None = Field(
        title="Sort Results",
        default=[("score", SortDirection.desc)],
    )
```

… will produce a schema definition like this:

```json
{
  "anyOf": [
    {
      "items": {
        "prefixItems": [
          {
            "type": "string"
          },
          {
            "$ref": "#/components/schemas/SortDirection"
          }
        ],
        "type": "array",
        "maxItems": 2,
        "minItems": 2
      },
      "type": "array"
    },
    {
      "type": "null"
    }
  ],
  "title": "Sort Results",
  "default": [
    [
      "score",
      "desc"
    ]
  ],
```
